### PR TITLE
Implementing discord Timestamps

### PR DESCRIPTION
**Pull request by AdhTri-001**
In this pull request I aimed to replace `moment().humanize()` to discord Timestamp that prevents blargbot to calculate humanized time and these also **Dynamic** with time in discord UI. This is how they look:
image

### CODE REVIEW

File: src/dcommands/ban.js
Comment (RagingLink) [inline src/dcommands/ban.js:78]:
```suggestion
            let unban_at = moment().add(duration).unix();

```

File: src/dcommands/ban.js
Comment (RagingLink) [inline src/dcommands/ban.js:87]:
```suggestion
                    endtime: r.epochTime(unban_at),

```

File: src/dcommands/mute.js
Comment (RagingLink) [inline src/dcommands/mute.js:115]:
```suggestion
                                let unmute_at = moment().add(duration).unix();

```

File: src/dcommands/mute.js
Comment (RagingLink) [inline src/dcommands/mute.js:125]:
```suggestion
                                        endtime: r.epochTime(unmute_at),

```

File: src/dcommands/remind.js
Comment (RagingLink) [inline src/dcommands/remind.js:60]:
```suggestion
                content: `:alarm_clock: Hi, <@${args.user}>! You asked me to remind you about this <t:${moment(args.starttime).unix()}:R> :
```

File: src/dcommands/remind.js
Comment (RagingLink) [inline src/dcommands/remind.js:67]:
```suggestion
            bu.sendDM(args.user, `:alarm_clock: Hi! You asked me to remind you about this <t:${moment(args.starttime).unix()}:R> :
```

File: src/dcommands/timer.js
Comment (RagingLink) [inline src/dcommands/timer.js:35]:
```suggestion
            content: `:alarm_clock: *Bzzt!* <@${args.user}>, the timer you set <t:${moment(args.starttime).unix()}:R> has gone off! *Bzzt!* :alarm_clock:`,
```
The discord timestamp can also be added to the message when a timer has been created o/ Pretty much identical to the `b!remind` implementation

**Review by RagingLink — CHANGES_REQUESTED** [review]

**Comment by AdhTri-001** [comment]
Thanks for fixing those.

File: src/dcommands/remind.js
Comment (danny-may) [inline src/dcommands/remind.js:54]: Remove the r.epochTime here too

**Comment by HunteRoi** [comment]
@AdhTri-001 First of all, thank you for your contribution!

However I think there is a misunderstanding here. @RagingLink and @Titansmasher only show how your code can be improved. Nothing has been changed nor fixed on your PR. It's actually awaiting _your_ changes. Can you please make a move so that we can make a second review of your PR before merging it?

Thank you again for your work! Glad to see you here :eyes:

File: src/dcommands/remind.js
Comment (AdhTri-001) [reply src/dcommands/remind.js]: Done 👍

**Comment by AdhTri-001** [comment]
Thanks @HunteRoi for telling that 😅

**Review by RagingLink — CHANGES_REQUESTED** [review]
Left a few comments o/

**Comment by AdhTri-001** [comment]
Bruh Idk what I did lol sorry :v)

**Comment by HunteRoi** [comment]
@AdhTri-001 by any chance, is it possible that you describe what is the objective of this PR ?

**Comment by AdhTri-001** [comment]
> @AdhTri-001 by any chance, is it possible that you describe what is the objective of this PR ?

[My try](https://github.com/blargbot/blargbot/pull/275#issue-706229060)

**Review by AdhTri-001 — COMMENTED** [review]
Just found the review option Aha!

File: src/dcommands/remind.js
Comment (HunteRoi) [inline src/dcommands/remind.js:54]: Can you externalize the moment calculation out of the string interpolation if possible? Look above at line 52, it's the same value used over there 😄

File: src/dcommands/remind.js
Comment (HunteRoi) [inline src/dcommands/remind.js:67]: Please use a scoped variable like before in order to avoid calculating the value at other places. Keep it simple : create a `duration` variable that stores the unix value of the time, and use this duration in the message with string interpolation!

File: src/dcommands/timer.js
Comment (HunteRoi) [inline src/dcommands/timer.js:35]: That'd be the same comment as for above changes: would it be possible to keep the value calculation outside of the string interpolation?

Thanks a lot!

**Review by HunteRoi — CHANGES_REQUESTED** [review]
Thank you again for your work. I believe [there are other files](https://github.com/blargbot/blargbot/search?q=.humanize%28true%29) where we send datetimes to Discord as a message content. If you have the time to do so, maybe you could take a look at these as well?

Anyway, here are a few comments of my own. Have a safe day!

File: src/dcommands/remind.js
Comment (RagingLink) [reply src/dcommands/remind.js:54]: I commented the same thing 🤔 weird

**Comment by AdhTri-001** [comment]
```js
if (isMember) {
            message = `\`\`\`prolog
${bu.padLeft('User', 19)} : ${userToGet.user.username}#${userToGet.user.discriminator}
${bu.padLeft('Username', 19)} : ${userToGet.user.username}
${bu.padLeft('Nickname', 19)} : ${userToGet.nick}
${bu.padLeft('Discriminator', 19)} : ${userToGet.user.discriminator}
${!userToGet.user.bot ? `${bu.padLeft('Account Type', 19)} : User` : `${bu.padLeft('Account Type', 19)} : Bot`}
${bu.padLeft('ID', 19)} : ${userToGet.user.id}
${bu.padLeft('Allowed Permissions', 19)} : ${userToGet.permissions.allow}
${bu.padLeft('Denied Permissions', 19)} : ${userToGet.permissions.deny}
${bu.padLeft('Avatar URL', 19)} : ${userToGet.user.avatarURL}
${userToGet.game == null ? `Not playing anything` : `Currently ${userToGet.game.type != null && userToGet.game.type > 0 ? 'streaming' : 'playing'} ${userToGet.game.name}`}
\`\`\`
\`${bu.padleft('Created At', 19)} :\` <t:${moment(userToGet.user.createdAt).unix()}> (<t:${moment(userToGet.user.createdAt).unix()}:R>)
\`${bu.padleft('Joined At', 19)} :\` <t:${moment(userToGet.joinedAt).unix()}> (<t:${moment(userToGet.joinedAt).unix()}:R>)`;
```
Thinking about this. Should I include this in `user.js`?

File: src/dcommands/remind.js
Comment (AdhTri-001) [reply src/dcommands/remind.js:54]: Aight

File: src/dcommands/timer.js
Comment (AdhTri-001) [reply src/dcommands/timer.js:36]: Sorry for late response, wasn't able to see these reviews before, as I was at phone :P
Anyways I did what u asked for, pls checkout commits below. Also suggest me about [this.](https://github.com/blargbot/blargbot/pull/275#issuecomment-896046169)

**Comment by RagingLink** [comment]
Not quite sure how that'd look tbh. I fear it might look a bit off outside the code block 🤔
Perhaps it'd be worth it to consider converting the output to an embed? Don't feel like that'd belong in this pr tbh

**Comment by AdhTri-001** [comment]
Was thinking the same. Embedding the command is much better choice

File: src/dcommands/remind.js
Comment (RagingLink) [inline src/dcommands/remind.js:61]: Please define this constant above `if (args.channel)` so it's defined and so it can be used in the DM message

File: src/dcommands/remind.js
Comment (RagingLink) [inline src/dcommands/remind.js:68]: use `endUnix` once defined.

**Review by RagingLink — CHANGES_REQUESTED** [review]

File: src/dcommands/remind.js
Comment (AdhTri-001) [reply src/dcommands/remind.js]: Done

**Comment by AdhTri-001** [comment]
Var name should be startUnix got bit confused there lol.

File: src/dcommands/time.js
Comment (HunteRoi) [inline src/dcommands/time.js:46]: I don't think these changes should be part of this PR. These are not related to implementing Discord timestamps. Please introduce them in another PR.

File: src/dcommands/timer.js
Comment (HunteRoi) [reply src/dcommands/timer.js:36]: Would it be possible to use a variable wherever you need to display something that has to be calculcated/instanciated before displayed? 
Here's an example: 

> image

Can you make the changes please?

File: src/events/guildMemberAdd.js
Comment (HunteRoi) [inline src/events/guildMemberAdd.js:52]: I know the code was already like that but is it possible to improve it by creating a variable to contain the value of `moment(member.user.createdAt).unix()` and call this variable in the string interpolation please?

**Review by HunteRoi — CHANGES_REQUESTED** [review]
We're close to the end 😄 

Talking about [this](https://github.com/blargbot/blargbot/pull/275#issuecomment-896046169), I agree with @RagingLink : it is not pretty to have the information outside of the code block and it's not possible to have the Discord timestamp formatting style in the codeblock. So the best way to go is to move to an embed. But yeah this has nothing to do with this PR, so that'd be great to have another one for this change!

File: src/events/guildMemberAdd.js
Comment (AdhTri-001) [reply src/events/guildMemberAdd.js]: Done

**Comment by RagingLink** [comment]
Looks good to me, though I'll do a test today when I have the time to make sure :)

File: src/structures/bbtag/Engine.js
Comment (AdhTri-001) [reply src/structures/bbtag/Engine.js:236]: May I also remove that in? That in in the text looks kinda sus.

Comment (AdhTri-001) [reply src/structures/bbtag/Engine.js:236]: Please try again in 7 minutes later

Comment (RagingLink) [reply src/structures/bbtag/Engine.js:236]: Totally forgot that, yes please.

**Review by RagingLink — APPROVED** [review]

**Review by HunteRoi — APPROVED** [review]

**Pull request merged**
